src/utils/hashStackTrace.ts

```diff
@@ -0,0 +1,24 @@
+import { createHash } from "crypto";
+
+/** Matches any line containing 'tsserver.js' followed by line numbers. */
+const serverLinePattern = /tsserver\.js:(\d+)(?::(\d+))?/i;
+
+export function getHash(methods: string[]): string {
+    const lines = methods.join("\n");
+    return createHash("md5").update(lines).digest("hex");
+}
+
+export function getHashForStack(stack: string): string {
+    const stackLines = stack.split(/\r?\n|\r/);
+
+    const errorMessage = stackLines[1];
+
+    // We will only match methods that contains tsserver. Everything else is ignored.
+    return getHash([errorMessage, ...stackLines.filter(l => serverLinePattern.exec(l))]);
```

Comment:
Out of curiosity, why does this matter? What lines create "noise" that we need to filter out any unrelated method names?

Reply:
Good question, I don't think it matters for this scenario. This code was copied from the errors coming from telemetry (where it matters).

Removed, as it is not necessary.